app/services/checksum_validator.rb

```diff
@@ -92,141 +87,202 @@ def moab_validator
+    def parse_verification_subentity(subentity)
+      add_cv_result_for_modified_xml(subentity) if subentity.subsets[MODIFIED]
+      add_cv_result_for_additions_in_xml(subentity) if subentity.subsets[ADDED]
+      add_cv_result_for_deletions_in_xml(subentity) if subentity.subsets[DELETED]
     end
-  rescue Errno::ENOENT
-    absent_from_moab_data = { manifest_file_path: latest_signature_catalog_path,
-                              file_path: signature_catalog_entry_path(entry) }
-    results.add_result(AuditResults::FILE_NOT_IN_MOAB, absent_from_moab_data)
-  end
 
-  # @return [String]
-  def signature_catalog_entry_path(entry)
-    @signature_catalog_entry_paths ||= {}
-    @signature_catalog_entry_paths[entry] ||= "#{object_dir}/#{entry.storage_path}"
-  end
+    def add_cv_result_for_modified_xml(subentity)
+      subentity.subsets.dig(MODIFIED, FILES).each_value do |details|
+        results.add_result(
+          AuditResults::MOAB_FILE_CHECKSUM_MISMATCH,
+          file_path: "#{subentity.details['other']}/#{details['basis_path']}",
+          version: subentity.details['basis']
+        )
+      end
+    end
 
-  # @return [String]
-  def latest_signature_catalog_path
-    @latest_signature_catalog_path ||= latest_moab_version.version_pathname.join(MANIFESTS, 'signatureCatalog.xml').to_s
-  end
+    def add_cv_result_for_additions_in_xml(subentity)
+      subentity.subsets.dig(ADDED, FILES).each_value do |details|
+        results.add_result(
+          AuditResults::FILE_NOT_IN_MANIFEST,
+          file_path: "#{subentity.details['other']}/#{details['other_path']}",
+          manifest_file_path: "#{subentity.details['other']}/#{MANIFESTS_XML}"
+        )
+      end
+    end
 
-  # @return [Array<SignatureCatalogEntry>]
-  def latest_signature_catalog_entries
-    @latest_signature_catalog_entries ||= latest_moab_version.signature_catalog.entries
-  rescue Errno::ENOENT, NoMethodError # e.g. latest_moab_version.signature_catalog is nil (signatureCatalog.xml does not exist)
-    results.add_result(AuditResults::SIGNATURE_CATALOG_NOT_IN_MOAB, signature_catalog_path: latest_signature_catalog_path)
-    []
-  rescue Nokogiri::XML::SyntaxError => e
-    results.add_result(AuditResults::INVALID_MANIFEST, manifest_file_path: latest_signature_catalog_path, addl: e.inspect)
-    []
+    def add_cv_result_for_deletions_in_xml(subentity)
+      subentity.subsets.dig(DELETED, FILES).each_value do |details|
+        results.add_result(
+          AuditResults::FILE_NOT_IN_MOAB,
+          file_path: "#{subentity.details['other']}/#{details['basis_path']}",
+          manifest_file_path: "#{subentity.details['other']}/#{MANIFESTS_XML}"
+        )
+      end
+    end
   end
 
-  def paths_from_signature_catalog
-    @paths_from_signature_catalog ||= latest_signature_catalog_entries.map { |entry| signature_catalog_entry_path(entry) }
-  end
+  # Validates files on disk against the signature catalog
+  class SignatureCatalogValidator
```

Comment:
I think this class is a great idea - but I would like it to be in a separate file with its own tests.   Perhaps that is "phase 2" of this specific refactoring and will be in a separate PR?

Reply:
I dunno if this class might be used in other places?

Reply:
Me neither. If we find it is useful, it can be pulled into a separate class and decoupled from ChecksumValidator.

Reply:
would it be worth making a ticket to find out?